client/src/components/WaitingRoom/index.js

```diff
@@ -0,0 +1,97 @@
+import React, { Component } from 'react';
+import { withFirebase } from '../Firebase'
+import { compose } from 'recompose';
+import { withAuthorization } from '../Session';
+
+class WaitingRoomBase extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      players: [],
+      gameId: '',
+      started: false
+    };
+
+    this.enterGame = this.enterGame.bind(this);
+  }
+
+  componentDidMount() {
+    const { id } = this.props.match.params;
+    const gameInstance = this.props.firebase.game(id);
+
+    // Listens for changes in the database for player list & status of whether the game has started
+
+    const unsubscribe = gameInstance.get()
```

Comment:
consts appear to be block-scoped so i'm not sure this can be referenced by componentWillUnmount. Have you tested these methods to make sure they're working?